xarray/tests/test_dataarray.py

```diff
@@ -3194,6 +3194,26 @@ def test_align_str_dtype(self) -> None:
         assert_identical(expected_b, actual_b)
         assert expected_b.x.dtype == actual_b.x.dtype
 
+    def test_align_exact_vs_strict(self) -> None:
+        xda_1 = xr.DataArray([1], dims="x1")
+        xda_2 = xr.DataArray([1], dims="x2")
+
+        # join='exact' passes
+        aligned_1, aligned_2 = xr.align(xda_1, xda_2, join="exact")
+        assert aligned_1 == xda_1
+        assert aligned_2 == xda_2
```

Comment:
The existing behavior does seem surprising.

Is it only an issue for arrays with 1 element though? Are there other examples of this behavior?

(Another option would be refining `exact` if that's not the behavior we want...)

Reply:
> The existing behavior does seem quite surprising. Is it only an issue with 1D arrays?

I added a test with 2D arrays, they are concerned too

What I can do as a simple test is to systematically transform 'exact' to 'strict' when entering the alignment logic and pushing this logic as a separate draft PR ("Reinforce alignment checks when join='strict'") to see the failing tests.